packages/shared/src/variants/s_fractional.ts

```diff
@@ -18,9 +18,10 @@ import {
   DefaultBoardState,
   MulticolorStone,
 } from "../lib/board_types";
+import { examineGroup } from "../lib/group_utils";
 
 declare type Color = string;
-declare type PlacementColors = [Color, Color] | [];
+declare type PlacementColors = Color[] & ([Color, Color] | []);
```

Comment:
I changed this type to enable calling `PlacementColors.includes(string)`, but I just realised that this also allows developers to call `PlacementColors.push(string)`. So I guess the typing is not ideal here either way.

Reply:
I think the idea behind tuple is that you should know what each member represents. So instead of:

```
color.includes("black");
```

You should be able to do something like:

```
// 0th element represents the "primary" color
color[0] === "black"

// or handling empty spaces
color.length && color[0] === black
```

Reply:
The empty array in `[] | [string, string]` makes it so we can't call `includes(string)`, because it's interpreted as `never[]`. If we first check whether `array.length === 0`, then typescript understands, and allows the call to 'includes(string)'.

I'll reconsider my future type choices with this in mind, but I think it's ok to leave it like this for now.

Reply:
Ohh I see now. I thought it was the other part causing issues!

I agree keeping includes makes sense even if type system is complaining.